bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java

```diff
@@ -0,0 +1,167 @@
+import org.jetbrains.annotations.Nullable;
+
+import com.adobe.cq.wcm.core.components.commons.link.Link;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Simple implementation for resolving and validating links from model's resources.
+ * This is a Sling model that can injected into other models using the <code>@Self</code> annotation.
+ */
+@Model(adaptables=SlingHttpServletRequest.class)
+public class LinkHandler {
+
+    /**
+     * List of allowed/supported values for link target.
+     * <code>_self</code> is used in the edit dialog but not listed as allowed here as we do not
+     * want to render a target attribute at all when <code>_self</code> is selected.
+     */
+    private static final Set<String> VALID_LINK_TARGETS = ImmutableSet.of("_blank", "_parent", "_top");
+
+    
+    @Self
+    private SlingHttpServletRequest request;
+    @ScriptVariable
+    private PageManager pageManager;
+    
+    /**
+     * Resolves a link from the properties of the given resource.
+     * @param resource Resource
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@NotNull Resource resource) {
+        return getLink(resource, PN_LINK_URL);
+    }
+
+    /**
+     * Resolves a link from the properties of the given resource.
+     * @param resource Resource
+     * @param linkURLPropertyName Property name to read link URL from.
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@NotNull Resource resource, String linkURLPropertyName) {
+        ValueMap props = resource.getValueMap();
+        String linkURL = props.get(linkURLPropertyName, String.class);
+        String linkTarget = props.get(PN_LINK_TARGET, String.class);
+        return getLink(linkURL, linkTarget);
+    }
+
+    /**
+     * Builds a link pointing to the given target page.
+     * @param page Target page
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@Nullable Page page) {
+        if (page == null) {
+            return getInvalid();
+        }
+        String linkURL = getPageLinkURL(page);
+        return new LinkImpl(linkURL, null, page);
+    }
+
+    /**
+     * Builds a link with the given Link URL and target.
+     * @param linkURL Link URL
+     * @param target Target
+     * @return Link may be invalid, but is never null
+     */
+    public @NotNull Link getLink(@Nullable String linkURL, @Nullable String target) {
+        String resolvedLinkURL = validateAndResolveLinkURL(linkURL);
+        String resolvedLinkTarget = validateAndResolverLinkTarget(target);
+
+        Page targetPage = pageManager.getPage(linkURL);
+
+        return new LinkImpl(resolvedLinkURL, resolvedLinkTarget, targetPage);
+    }
+
+    /**
+     * Returns an invalid link.
+     * @return Invalid link, never null
+     */
+    public @NotNull Link getInvalid() {
```

Comment:
I'd rename it `getInvalidLink` to be consistent with the other `getLink` methods.

Reply:
yes - alternative 1:
use method names like `get(...)` and `getInvalid()`
alternative 2:
use method names like `getLink(...)` and `getInvalidLink()`
for consistency. i slightly tend to alternative 1.

Reply:
I much prefer well describing method names (alternative 2)

Reply:
I think `LinkHandler#getLink` and `LinkHandler#getInvalidLink` make for more self-explanatory names. Done in acf1edb